llvm/lib/CAS/HashMappedTrie.cpp

```diff
@@ -464,10 +462,11 @@ void ThreadSafeHashMappedTrieBase::destroyImpl(function_ref<void (void *)> Destr
         if (auto *Content = dyn_cast_or_null<TrieContent>(Slot.load()))
           Destructor(Content->getValuePointer());
 
-  // Destroy the subtries using explicit iteration here to avoid recursive
-  // destructor calls. Incidentally, this destroys them in reverse order of
-  // creation (breadth-first, deepest first).
-  std::unique_ptr<TrieSubtrie> Trie = Impl->Root.Next.take();
-  while (Trie)
-    Trie = Trie->Next.take();
+  // Destroy the subtries.
+  TrieSubtrie *Trie = Impl->Root.Next;
+  while (Trie) {
```

Comment:
I think the "Incidentally, this destroys them in reverse order..." is still accurate, and might also be valuable in the future. WDYT?